libs/product/src/reducers/composite-product-entities/composite-product-entities.reducer.ts

```diff
@@ -87,12 +85,16 @@ function getDefaultOption(item: DaffCompositeProductItem): DaffCompositeProductE
 			qty: 1
 		}
 	} else {
-		return item.required && isOptionInStock(item.options[0]) ? 
-			{ value: item.options[0].id, qty: 1 } :
+		return item.required && getFirstInStockOptionIndex(item) > -1 ? 
```

Comment:
store `getFirstInStockOptionIndex(item)` in a local var.

Reply:
done